src/devices/machine/am9517a.h

```diff
@@ -46,6 +46,8 @@ class am9517a_device :  public device_t,
 						public device_execute_interface
 {
 	friend class pcxport_dmac_device;
+	friend class ps2_dma_device;
+	friend class ibm72x7377_device;
```

Comment:
You *really* shouldn’t be needing to make another device a friend like this – this totally breaks encapsulation.